vespalib/src/vespa/vespalib/stllike/string.h

```diff
@@ -200,7 +216,25 @@ class small_string
             free(buffer());
         }
     }
-    small_string& operator= (const small_string &rhs) {
+    small_string& operator= (small_string && rhs) noexcept {
+        reset();
+        _sz = rhs._sz;
+        _bufferSize = rhs._bufferSize;
+        if (rhs.isAllocated()) {
+            _buf = rhs._buf;
+            rhs._buf = rhs._stack;
+            rhs._sz = 0;
+            rhs._bufferSize = sizeof(rhs._stack);
+            rhs._stack[0] = 0;
+        } else {
+            _buf = _stack;
+            memcpy(_stack, rhs._stack, sizeof(_stack));
+            rhs._sz = 0;
+            rhs._stack[0] = 0;
+        }
```

Comment:
Could consider factoring out the shared code between the move ctor and move assignment operator into a separate function

Reply:
Fixed